src/share/classes/com/sun/javatest/regtest/tool/Tool.java

```diff
@@ -2196,12 +2195,20 @@ private void addEnvVars(Map<String, String> table, Map<String, String> sysEnv, L
         }
     }
 
-    private void addEnvVars(Map<String, String> table, Map<String, String> sysEnv, Predicate<Map.Entry<String, String>> filter) {
-        sysEnv.entrySet().stream()
-                .filter(filter)
+    // This method streams over a case-sensitive set of entries, returned by
+    // System.getenv(String); custom name filters can be implemented in a
+    // case-insensitive manner.
+    private void addEnvVarsByName(Map<String, String> table, Predicate<String> nameFilter) {
+        System.getenv().entrySet().stream()
+                .filter(e -> nameFilter.test(e.getKey()))
                 .forEach(e -> table.put(e.getKey(), e.getValue()));
     }
 
+    // 7903515: Use case-insensitive System.getenv(String) here.
+    private static String getCaseInsensitiveEnvironmentVariableOrNull(String name) {
```

Comment:
Suggest removing the `CaseInsensitive` part of the name, since it is not true on non-Windows systems.